ifrs17/Utils/Queries.ipynb

```diff
@@ -450,23 +450,46 @@
         {
             "cell_type": "code",
             "source": [
-                "public static async Task<T[]> QueryPartitionedDataAsync<T,P>(this IWorkspace workspace, IDataSource dataSource, Guid targetPartition, Guid defaultPartition, string format)",
+                "public static bool GetIsRelaxed<T>(string format)",
+                "\n    where T : IPartitioned",
+                "\n{",
+                "\n    if(typeof(ImportFormats).GetFields().Select(x => x.Name).Contains(format)) //ifrs17 ImportFormats",
+                "\n        return ((format != ImportFormats.Cashflow && typeof(T).Name == nameof(IfrsVariable)) ||",
+                "\n               (format == ImportFormats.Cashflow && typeof(T).Name == nameof(RawVariable)));",
+                "\n    ",
+                "\n    //Override or simple expansion allows to extend this concept to other Formats",
+                "\n    return true;",
+                "\n    //Scenario == null (BE) => workspace",
+                "\n    //Scenario == ",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static async Task<T[]> QueryPartitionedDataAsync<T,P>(this IWorkspace workspace, IDataSource dataSource, ",
+                "\n                                                                  Guid targetPartition, Guid defaultPartition, ",
+                "\n                                                                  string format,",
+                "\n                                                                  Expression<Func<T, bool>> filterExpression = null)",
                 "\n    where T : IPartitioned",
                 "\n    where P : IPartition",
                 "\n{",
-                "\n    var isRelaxed = ((format != ImportFormats.Cashflow && typeof(T).Name == nameof(IfrsVariable)) ||",
-                "\n                    (format == ImportFormats.Cashflow && typeof(T).Name == nameof(RawVariable)));",
+                "\n    //If Ifrs17 format",
+                "\n    var isRelaxed = GetIsRelaxed<T>(format);",
                 "\n",
-                "\n    var variablesFromWorkspace = await workspace.LoadPartitionedDataAsync<T,P>(targetPartition);",
```

Comment:
LoadPartitionedDataAsync this method throws an error. (No overload for method 'LoadPartitionedDataAsync' takes 2 arguments).